### PR TITLE
chore(currency): update aggregators

### PR DESCRIPTION
**Pull request by alexandre-abrioux**
## Description of the changes

- Updated the command `yarn chainlinkPath mainnet` to use the official Chainlink price feed API
- Fixed out-of-memory error when using the command on networks with a lot of blocks like Arbitrum
- Updated the aggregator files for Ethereum mainnet and Arbitrum

### CODE REVIEW

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 1]
```diff
@@ -56,37 +54,35 @@ class ChainlinkConversionPathTools {
 
     this.creationBlockNumber = chainlinkConversionPath.getCreationBlockNumber(this.network);
 
-    this.maxRange = options?.maxRange || 5000;
+    this.maxRange = options?.maxRange || 1000000;
```
Comment (alexandre-abrioux) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:57]: Testing with this value: RPCs can easily handle this amount of block for such contracts with a very low amount of transactions.

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 2]
```diff
@@ -56,37 +54,35 @@ class ChainlinkConversionPathTools {
...
    * Retrieves all the aggregators
    */
   public async getAggregators(): Promise<Record<string, Record<string, string>>> {
     const lastBlock = await this.provider.getBlockNumber();
-    const chunks = chunk(
-      Array(lastBlock - this.creationBlockNumber)
-        .fill(0)
-        .map((_, i) => this.creationBlockNumber + i),
-      this.maxRange,
-    );
```
Comment (alexandre-abrioux) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:72]: This piece of code is problematic and results in an out-of-memory error on networks with many blocks like Arbitrum. I replaced it with a simple while loop.

File: packages/toolbox/src/chainlinkConversionPathTools.ts [hunk 3]
```diff
@@ -56,37 +54,35 @@ class ChainlinkConversionPathTools {
...
-      chunks,
-      (blocks) => {
-        console.error(`Fetching logs from ${blocks[0]} to ${blocks[blocks.length - 1]}`);
-        return retry(this.chainLinkConversionPath.queryFilter.bind(this.chainLinkConversionPath), {
+    const logs = [];
+    while (currentBlock <= lastBlock) {
+      const nextBlock = currentBlock + this.maxRange;
+      console.error(
+        `Fetching logs from ${currentBlock} to ${nextBlock} (progress: ${Math.round(
+          (currentBlock * 100) / lastBlock,
```
Comment (alexandre-abrioux) [inline packages/toolbox/src/chainlinkConversionPathTools.ts:73]: added a percent progress log ✨

File: packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts [hunk 4]
```diff
@@ -29,35 +17,34 @@ export type Aggregator = {
...
     throw new Error(
       `network ${network} not supported by feed provider. Is it supported by Chainlink?`,
     );
   }
-  const { data } = await axios.get<Record<string, Feed>>(
-    'https://cl-docs-addresses.web.app/addresses.json',
+
+  const { data: feeds } = await axios.get<Feed[]>(
+    `https://reference-data-directory.vercel.app/feeds-${feedName}.json`,
   );
```
Comment (alexandre-abrioux) [inline packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:42]: In this file, I changed the API for fetching feeds. I don't know where the previous URL came from, I can't find any doc on it, and it looked outdated. For instance, there was no Arbitrum feed. The new URL is more or less the official "Chainlink feed API": it is the API Chainlink uses on their feed list page [https://docs.chain.link/data-feeds/price-feeds/addresses](https://docs.chain.link/data-feeds/price-feeds/addresses)

**Review by yomarion — APPROVED** [review]

File: packages/toolbox/src/commands/chainlink/addAggregator.ts [hunk 5]
```diff
@@ -79,9 +79,9 @@ export const handler = async (args: Options): Promise<void> => {
   }
   if (!aggregator.startsWith('0x')) {
     const aggregators = await getAllAggregators(network);
-    const newAggregator = aggregators.find((x) => x.pair === aggregator);
+    const newAggregator = aggregators.find((x) => x.name === aggregator);
     assert(newAggregator, `aggregator ${aggregator} not found`);
-    aggregator = newAggregator.proxy;
+    aggregator = newAggregator?.proxyAddress;
```
Comment (olivier7delf) [inline packages/toolbox/src/commands/chainlink/addAggregator.ts:84]: I would specify aggregatorAddress or rename aggregators into aggregatorFeeds

**Review by olivier7delf — APPROVED** [review]

File: packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts [hunk 6]
```diff
@@ -29,35 +17,34 @@ export type Aggregator = {
...
-  rinkeby: ['ethereum', 'Rinkeby Testnet'],
-  fantom: ['fantom', 'Fantom Mainnet'],
-  matic: ['polygon', 'Polygon Mainnet'],
-  xdai: ['gnosis-chain', 'Gnosis Chain Mainnet'],
-  bsc: ['bnb-chain', 'BNB Chain Mainnet'],
-  avalanche: ['avalanche', 'Avalanche Mainnet'],
-  optimism: ['optimism', 'Optimism Mainnet'],
-  'arbitrum-one': ['arbitrum', 'Arbitrum Mainnet'],
-  moonbeam: ['moonbeam', 'Moonbeam Mainnet'],
+  mainnet: ['mainnet', 'Ethereum Mainnet'],
```
Comment (benjlevesque) [inline packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:20]: FYI this data is now available on the chains API: https://api.request.finance/currency/chains/mainnet, see `priceFeeds`

File: packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts (hunk 4 above)
Comment (benjlevesque) [reply packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:42]: the previous address came from retro engineering their page :D
nothing documented

**Review by benjlevesque — APPROVED** [review]

**Review by leoslr — APPROVED** [review]

File: packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts (hunk 6 above)
Comment (alexandre-abrioux) [reply packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:20]: @benjlevesque The values are different with this other API 😕

Comment (benjlevesque) [reply packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:20]: OK. maybe we can update the API values to match that

Comment (benjlevesque) [reply packages/toolbox/src/commands/chainlink/aggregatorsUtils.ts:20]: but not needed now

File: packages/toolbox/src/commands/chainlink/addAggregator.ts (hunk 5 above)
Comment (alexandre-abrioux) [reply packages/toolbox/src/commands/chainlink/addAggregator.ts:84]: @olivier7delf I won't refactor the command in this PR but I'm creating a follow-up here: https://github.com/RequestNetwork/requestNetwork/pull/1195

**Pull request merged**